### PR TITLE
Fix bug where `TileFrame` would not pass through the keys of its chil…

### PR DESCRIPTION
**Pull request by ghsolomon**
…dren

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/layout/TileFrame.ts [hunk 1]
```diff
@@ -101,7 +101,8 @@ export const [TileFrame, tileFrame] = hoistCmp.withFactory({
             childrenArr.map((item, idx) => box({
                 style: localModel.getTileStyle(idx),
                 className: 'xh-tile-frame__tile',
-                item
+                item,
+                key: isObject(item) && 'key' in item ? item.key : undefined
             })):
```
Comment (cnrudd) [inline cmp/layout/TileFrame.ts:106]: I think an explanation in the form of a comment is needed here, to explain why passing the item keys to the box is necessary.  What issues is this solving/avoiding?

**Comment by lbwexler** [comment]
I am looking at this too.   Think it makes sense, but want to understand
the motivation here

**Comment by ghsolomon** [comment]
Children passed to `TileFrame` were being individually wrapped in container elements which did not themselves inherit the keys of their child components. 

This caused Children with keys to be fully re-rendered unnecessarily, whenever their index changed.  In addition to this negatively impacting performance by triggering extra re-renders, it caused any transient uncontrolled state (such as an uncontrolled `input`) to be lost.

**Comment by cnrudd** [comment]
I think another note explaining the importance of adding 'key' to this component's children is warranted, at the top of the component, inside the existing 'intro' comment.

**Comment by cnrudd** [comment]
> I am looking at this too.   Think it makes sense, but want to understand
> the motivation here
> […](#)

Greg put together some really great examples in his local sandbox of how the component fails without this change.  Might be worth a screenshare.

**Comment by lbwexler** [comment]
No worries -- I am convinced

**Pull request merged**